corehq/apps/locations/adjacencylist.py

```diff
@@ -0,0 +1,273 @@
+from __future__ import absolute_import
+from django.contrib.postgres.fields.array import ArrayField
+from django.db.models import CharField, IntegerField
+from django.db.models.aggregates import Max
+from django.db.models.expressions import F, Func, Value
+from django.db.models.query import Q, QuerySet
+from django_cte import With
+from mptt.models import MPTTModel, TreeManager
+
+from .queryutil import ComparedQuerySet, TimingContext
+
+int_field = IntegerField()
+str_array = ArrayField(CharField())
+
+
+class StrArray(Func):
```

Comment:
Why is this in CamelCase, while the other `Func` subclasses are lowercase?

Reply:
I was thinking of this one as a type, while the others are functions, although on reflection that distinction is pretty fuzzy in this case. Also, the name `str_array` is already used above for the field instance.